cranelift/codegen/src/machinst/abi_impl.rs

```diff
@@ -163,25 +163,72 @@ pub enum ABIArg {
         /// Purpose of this arg.
         purpose: ir::ArgumentPurpose,
     },
+    /// Sentinel for representing no arg in `ValueRegs` wrapper.
```

Comment:
This is used by `ValueRegs`'s packed encoding -- it doesn't have a length field, instead using "invalid" sentinels for parts beyond the needed part-count. Perhaps my reuse of `ValueRegs` rather than something simpler like `Vec<ABIArgPart>` is the real issue -- in hindsight it's probably better to just do that!

Reply:
Changed -- this has now been refactored so that we have a `Vec` of `ABIArgSlot`s (old "parts") and don't overload `ValueRegs`.